ivadomed/utils.py

```diff
@@ -72,14 +68,17 @@ def unstack_tensors(sample):
     return list_tensor
 
 
-def generate_sha_256(context: dict, df: pd.DataFrame, file_lst: List[str]) -> None:
+def generate_sha_256(context: dict, df, file_lst: List[str]) -> None:
     """generate sha256 for a training file
 
     Args:
         context (dict): configuration context.
         df (pd.DataFrame): Dataframe containing all BIDS image files indexed and their metadata.
         file_lst (List[str]): list of strings containing training files
     """
+    from pandas import DataFrame
+    assert isinstance(df, DataFrame)
```

Comment:
PD.DataFrame type hinting in this case actually increase load time quite a bit so replaced here with a hard data type check assert. 

Suggestions welcome here.

Reply:
not a big fan of inline imports...but if it gets the job done...